app/src/main/java/sims/michael/gerritslackbot/model/ChangeMatcher.kt

```diff
@@ -29,10 +29,8 @@ data class ChangeMatcher(
     private val ChangeEvent.changeKindOrNull: String?
         get() = if (this is PatchSetCreatedEvent) patchSet.kind.toString() else null
 
-    private fun String.exactOrRegexMatch(pattern: String) = when {
-        pattern.startsWith("^") -> this.safeMatches(pattern)
-        else -> pattern.toLowerCase() in listOf("*", this.toLowerCase())
-    }
+    private fun String?.safeMatchesWildcardOrRegex(pattern: String) =
+            pattern == "*" || this?.safeMatches(pattern) ?: true
```

Comment:
The last boolean in this expression should be `false`. As it is, if a change subject is missing for some reason (null) then any subject pattern provided will match it. A missing subject should be matched only be `"*"`

Reply:
Good point.